From 5c20833a04071ee114e61691881edde5bae11a38 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:38:32 -0400
Subject: [virt] kvm/x86: Add comment on vcpu_enter_guest()'s return value

Message-id: <1406040016-3289-118-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86196
O-Subject: [RHEL7 PATCH v2 117/221] KVM: x86: Add comment on vcpu_enter_guest()'s return value
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Takuya Yoshikawa <yoshikawa_takuya_b1@lab.ntt.co.jp>

Giving proper names to the 0 and 1 was once suggested.  But since 0 is
returned to the userspace, giving it another name can introduce extra
confusion.  This patch just explains the meanings instead.

Signed-off-by: Takuya Yoshikawa <yoshikawa_takuya_b1@lab.ntt.co.jp>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit 9357d93952143b178fa9d1f5095b8f273b01a1f1)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index f9f1aa2..fc88300 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -5864,6 +5864,11 @@ static void vcpu_scan_ioapic(struct kvm_vcpu *vcpu)
  kvm_apic_update_tmr(vcpu, tmr);
 }
 
+/*
+ * Returns 1 to let __vcpu_run() continue the guest execution loop without
+ * exiting to the userspace.  Otherwise, the value will be returned to the
+ * userspace.
+ */
 static int vcpu_enter_guest(struct kvm_vcpu *vcpu)
 {
  int r;
-- 
1.7.1